comply with pep-0353 requirements

xanimo opened the pull request:
On Ubuntu 22.04 Jammy, the previous archive hosted by langerhans would not install on account of Python 3.10.5's requirement to define the PY_SSIZE_T_CLEAN macro prior to including <Python.h> for all # variants of formats (s#, y#, etc.) found in scryptmodule.c. This PR fixes that. 

References:
https://bugs.launchpad.net/ubuntu/+source/python-systemd/+bug/1963582
https://github.com/python/cpython/commit/569ca81adf0be92be8752f6cc6492117f9ef3c0b
https://peps.python.org/pep-0353/
https://github.com/dogecoin/dogecoin/issues/2656

patricklodder reviewed (commented):
Tested on x86_64 Ubuntu 22.04 - all good.

Minor request: could you please remove the special characters from the first commit message line? Maybe rewording to something like the below will also make it easier to understand in 10 years from now: 
```
comply with pep-0353 requirements

PEP-0353 compliance is needed for Python >= 3.10 to 
accept `#`-formatting. see: https://peps.python.org/pep-0353/
```

xanimo commented:
> Tested on x86_64 Ubuntu 22.04 - all good.
> 
> Minor request: could you please remove the special characters from the first commit message line? Maybe rewording to something like the below will also make it easier to understand in 10 years from now: 
> ```
> comply with pep-0353 requirements
> 
> PEP-0353 compliance is needed for Python >= 3.10 to 
> accept `#`-formatting. see: https://peps.python.org/pep-0353/
> ```

absolutely yes good call. thanks!

xanimo commented:
Updated. Forgive me for lack of imagination with rewording. Slow start/hectic day heh.

patricklodder reviewed (approved):
ACK - tested on:

- bionic / Python 3.6.9
- focal / Python 3.8.10
- jammy / Python 3.10.4

xanimo commented:
> ACK - tested on:
> 
>     * bionic / Python 3.6.9
> 
>     * focal / Python 3.8.10
> 
>     * jammy / Python 3.10.4

As per contributor guideline I will wait 24 hours to merge this unless I hear otherwise.

patricklodder commented:
It's imho better to merge the License statement from #2 before this - otherwise you'd be publishing code without a license.

xanimo commented:
> It's imho better to merge the License statement from #2 before this - otherwise you'd be publishing code without a license.

Would you mind if I merged the license and contributor guide? And yes good call was just indicating I had read it and can follow directions 😅

patricklodder commented:
I think we can give it some time - there is no real threatening urgency to those or this PR. Would be better to just take it easy, give other people time to read it.

xanimo commented:
> I think we can give it some time - there is no real threatening urgency to those or this PR. Would be better to just take it easy, give other people time to read it.

sgtm 🙂

xanimo committed:
comply with pep-0353 requirements

PEP-0353 compliance is needed for Python >= 3.10 to
accept `#`-formatting. see: https://peps.python.org/pep-0353/

patricklodder reviewed (approved):
re-ACK on 4c0f3fac
